bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/Property.java

```diff
@@ -280,7 +286,7 @@ public void createChannelFromAttribute() {
         }
         // Make sure we set the callback again which might have been nulled during an stop
         channelState.setChannelStateUpdateListener(this.callback);
-        return channelState.start(connection, scheduler, timeout);
+        return channelState.start(connection, scheduler, timeout).thenRun(() -> starting.set(false));
```

Comment:
From what I can tell the fix should be in ChannelState.start() instead. It isn't thread safe and it is no surprise that you are seeing the issues that you are seeing. Your efforts should be to make that method thread-safe and return ongoing futures if it called concurrently.
Fix that and I suspect it should solve your problem.

Try making these changes in ChannelState and see if it fixes your problem.
```
private CompletableFuture<@Nullable Void> future = CompletableFuture.completedFuture(null);
private final Object futureLock = new Object();

public CompletableFuture<@Nullable Void> start(MqttBrokerConnection connection, ScheduledExecutorService scheduler,
            int timeout) {
        sychronized(futureLock){
          // if the connection is still the same, the subscription is still present, otherwise we need to renew
          if ((hasSubscribed || !future.isDone()) && connection.equals(this.connection)) {
              return future;
          }
          hasSubscribed = false;
          
          this.connection = connection;

          if (StringUtils.isBlank(config.stateTopic)) {
              return CompletableFuture.completedFuture(null);
          }

          this.future = new CompletableFuture<>();
        }
        connection.subscribe(config.stateTopic, this).thenRun(() -> {
            hasSubscribed = true;
            logger.debug("Subscribed channel {} to topic: {}", this.channelUID, config.stateTopic);
            if (timeout > 0 && !future.isDone()) {
                this.scheduledFuture = scheduler.schedule(this::receivedOrTimeout, timeout, TimeUnit.MILLISECONDS);
            } else {
                receivedOrTimeout();
            }
        }).exceptionally(this::subscribeFail);
        return future;
    }
```

Reply:
Hi @cpmeister! Thanks for your suggestion, I've implemented it and it also works fine (I didn't expect anything else coming from you :wink:)!

@Hilbrand and @jochen314: what do you think? Should I commit the suggested changes?